packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -398,13 +477,51 @@ function WorklistDataGrid(props) {
     }
   };
 
-  const handleCellSelection = (cellRowIdColId, cellCoordinates) => {
-    selectCell(cellRowIdColId, cellCoordinates);
+  const handleCellSelection = (selectionDetails) => {
+    selectCell(selectionDetails);
   };
 
-  const handleRowSelection = (rowId, rowIndex, selectedCellCoordinates) => {
-    handleCellSelectionChange(null, null, selectedCellCoordinates);
-    selectRows(false, rowId, rowIndex);
+  const handleRowSelection = (selectionDetails) => {
+    handleCellSelectionChange(null, null, { row: selectionDetails.rowIndex, col: selectionDetails.columnIndex });
+    const selectionData = selectionDetails.selectedByKeyboard ? rowSelection.current.lastIndividuallySelectedRow : rowSelection?.current?.focussedRow;
+
+    if (!selectionDetails.multiSelect) {
+      if (hasSelectableRows) {
+        // regular click or space key
+        selectRows(false, selectionDetails.rowId, selectionDetails.rowIndex);
+      }
+    } else if (hasSelectableRows) {
+      // Shift key is pressed
+      if (selectionData?.anchorRow) {
```

Comment:
How does an anchor row let us know that the shift key has been pressed?

Reply:
The anchor row does not. It is the condition in the _if_ block that allows us to make the determination.